### ISSUE TITLE
0.41.0: find_library() change - can't find lib anymore

### ISSUE THREAD

**Issue by rhd** [mesonbuild/meson#1936]
This has me stumped.

I upgraded from meson 0.39.1 to 0.41.0 and `x264 = cc.find_library('x264')` doesn't work anymore.

```
$ cat /etc/ld.so.conf.d/local.so.conf 
/usr/local/lib
```

```
$ ls -l /usr/local/lib | grep x264
-rw-r--r-- 1 root root   1708882 May 28  2013 libx264.a
lrwxrwxrwx 1 root root        14 May 28  2013 libx264.so -> libx264.so.133
-rwxr-xr-x 1 root root   1384172 May 28  2013 libx264.so.133
```

I switch back to 0.39.1 and it works great.

Any clue what has happened?

**Comment by QuLogic** [comment]
Unrelated to the problem, but is there a reason you aren't using the pkg-config file?

**Comment by rhd** [comment]
@QuLogic yes, these dev boxes are a mess and they get managed by people scp'ing stuff around... thus, almost nothing is installed properly.

Just to verify to myself...
```
Found pkg-config: /usr/bin/pkg-config (0.23)

Meson encountered an error in file meson.build, line 103, column 0:
Native dependency 'x264' not found
```

`pkg-config --list-all | grep 264` returns nothing.

Thankfully `find_library` works pretty well.

**Comment by jpakkane** [comment]
By default /usr/local/lib/pkg-config is not in pkg-config's search path. You need to add it to `PKG_CONFIG_PATH`. That does not really help if someone has just scp'd the libs and headers without the .pc file, though ...

**Comment by nirbheek** [comment]
`ld.so.conf` is where the library loader looks at runtime, not where `ld` looks. However, we haven't changed much in `find_library` since 0.39.1. What's your `meson.build` file look like? Also can you paste the relevant part of `meson-log.txt`? For both 0.39.1 and 0.41.0.

**Comment by rhd** [comment]
meson.build
```
#x264
x264 = cc.find_library('x264')
```

meson is being called like this (it's checked into the repo):
```
CC="ccache gcc-4.8.1" CXX="ccache g++-4.8.1" python3 ./meson/meson.py --buildtype=debugoptimized build-m
```

meson-log.txt 0.41.0
```
unning compile:
Working directory:  /tmp/tmpdqstvf6h
Command line:  ccache gcc-4.8.1 /tmp/tmpdqstvf6h/testfile.c -O0 -Wl,--start-group -Wl,--start-group -lx264 -Wl,--end-group -Wl,--end-group -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmpdqstvf6h/output.exe 

Code:
 int main(int argc, char **argv) {
    return 0;
}
        
Compiler stdout:
 
Compiler stderr:
 /usr/bin/ld: may not nest groups (--help for usage)
collect2: error: ld returned 1 exit status


Meson encountered an error in file meson.build, line 103, column 0:
C library 'x264' not found
```

meson-log.txt 0.39.1
```
Running compile:
Working directory:  /tmp/tmpnxwoiefv
Command line:  ccache gcc-4.8.1 /tmp/tmpnxwoiefv/testfile.c -O0 -lx264 -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmpnxwoiefv/output.exe 

Code:
 int main(int argc, char **argv) {
    return 0;
}
        
Compiler stdout:
 
Compiler stderr:
 
Library x264 found: YES
```

Some interesting differences in the way x264 is checked.

**Comment by nirbheek** [comment]
I see the problem. Working on a patch.

**Comment by rhd** [comment]
@nirbheek thanks :+1:

**Comment by nirbheek** [comment]
Which version of `ld` is this, btw?

**Comment by rhd** [comment]
From redhat 6.5

```
$ ld --version
GNU ld version 2.20.51.0.2-5.36.el6 20100205
```

**Comment by nirbheek** [comment]
Fixed by #1960.
